rfd/0090-db-mfa-sessions.md

```diff
@@ -0,0 +1,175 @@
+---
+author: Gavin Frazar (gavin.frazar@goteleport.com)
+state: draft
+---
+ 
+# RFD 90 - Database MFA Sessions
+
+## Required Approvers
+- Engineering: Marek `@smallinsky`, Roman `@r0mant`
+- Product: Sasha `@klizhentas`, Xin `@xinding33`
+
+## What
+When per-session-MFA is enabled, we should not restrict database cert TTL to 1 minute.
+
+Instead, database cert TTL should be restricted to `max_session_ttl`, and the cert
+should be kept in-memory by a local proxy tunnel.
+
+## Why
+Database access via GUI client such as DataGrip requires the user to start a local proxy to connect to: `tsh proxy db --tunnel <db>`.
+
+To support per-session-MFA for the local proxy, we are adding a callback function called on each new connection to the local proxy.
+This callback function will check if the database certs need to be reissued to proxy the connection, and prompt for credentials/MFA if needed.
+
+Currently, database certs are issued with 1 minute ttl if per-session-MFA is enabled as described in [RFD 14](https://github.com/gravitational/teleport/blob/master/rfd/0014-session-2FA.md).
+This causes poor UX when the GUI client can establish a new connection at any time,
+and a user may be prompt once per minute for their MFA.
+
+We should remove this 1 minute restriction on database cert TTL to improve UX.
+
+## Details
+
+"Doesn't this just disable per-session-mfa for database access?" (My initial thinking)
+
+- No, not quite. Sessions are still limited by the lifetime of a local proxy process and an MFA prompt is always required to start these proxies when per-session-mfa is enabled.
+- This can "weaken" per-session-mfa security though. If a user finds that they would prefer the old behavior of 1-minute TTL certs, they can still achieve that through configuration using `max_session_ttl`.
```

Comment:
Is this true? If "disconnect on expired cert" is set, the user would be disconnected after 1 minute.

Reply:
`disconnect_expired_cert: true` also disconnects per-session-mfa certs after 1 minute, so there's no difference in behavior there. I just tested this with ssh and database access to be sure.